lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -327,29 +320,110 @@ class ProviderTest {
 
     @Test
     fun `get deploy transaction receipt gateway`() {
-        val request = gatewayProvider().getTransactionReceipt(deployTransactionHash)
+        // Mocked response of GOL2 test contract deploy
+        // Fetched from testnet using
+        // starknet get_transaction_receipt --hash 0x01a9d9e311ff31e27b20a7919bec6861dd6b603d72b7e8df9900cd7603200d0b
+        val httpService = mock<HttpService> {
+            on { send(any()) } doReturn HttpResponse(
+                true,
+                200,
+                """
+                    {
+                        "actual_fee": "0x0",
+                        "block_hash": "0x4e782152c52c8637e03df60048deb4f6adf122ef37cf53eeb72322a4b9c9c52",
+                        "block_number": 264715,
+                        "events": [],
+                        "execution_resources": {
+                            "builtin_instance_counter": {
+                                "bitwise_builtin": 0,
+                                "ecdsa_builtin": 0,
+                                "output_builtin": 0,
+                                "pedersen_builtin": 0,
+                                "range_check_builtin": 0
+                            },
+                            "n_memory_holes": 0,
+                            "n_steps": 41
+                        },
+                        "l2_to_l1_messages": [],
+                        "status": "ACCEPTED_ON_L1",
+                        "transaction_hash": "0x1a9d9e311ff31e27b20a7919bec6861dd6b603d72b7e8df9900cd7603200d0b",
+                        "transaction_index": 8
+                    }
+                """.trimIndent(),
+            )
+        }
```

Comment:
I wonder it wouldn't be better to also have same tests using testnet itself/replace these mocks with tests like that?

Reply:
It could be, but it also would mean that we would be susceptible to some random problems with tests. Not sure if we gain that much from additionally querying external service, but we can discuss

Reply:
@Solpatium

Reply:
I think having mocks are okay in cases like this. I just wondered if we prefer risk of mocks being outdated vs possible problems with querying testnet.

Reply:
I think mock in here makes sense, as long it is accompanied with command for getting it.